### DISCUSSION TITLE
Inertia Route Dynamically Added

### DISCUSSION THREAD

**Discussion by mikesaintsg** [adonisjs/core#4597] [General]
Inertia currently appends an inertia route `/inertia` but it's confusing when using with a starter kit and get an error that the home route `/` can't be found. 

It would be nice for it to check whether there is a home route already. If not, then make the inertia the home route, and if so, just use the inertia route `/inertia`

I can fix this quickly with a few lines and a test, just want to make sure it's worth a pull request, happy to provide

**Comment by RomainLanz** [comment]
Hey @mikesaintsg! 👋🏻 

We are happy to accept a PR for this.
In fact, I am wondering if we should even bother checking for the home route and not directly binding `/` for Inertia.

Most people install the starter kit directly, and they don't add Inertia in an already running application.

What are your thoughts @Julien-R44?

**Reply by Julien-R44** [reply]
Yes, that's what I thought. Maybe just setting the route to `/` is enough.
Because if we want to have a conditional addition, it won't be as simple as that, because today the route is added via `ts-morph`. So we have 2 solutions:
- Either we have to do some static analysis with ts-morph to determine whether the route is defined. Unreliable because it can be defined somewhere other than `start/routes.ts`.
- Run `node ace list:routes` from `configure.ts` to determine if the route is already taken.

so 2 solutions that are a bit of a pain to set up. I think I would rather not bother and just set the route to `/` yes

**Reply by thetutlage** [reply]
How about adding the route by default in the Inertia starter kit and then accepting a flag via `configure` command to skip registering routes.

**Reply by Julien-R44** [reply]
Yeah, seems to be the best approach!

**Reply by mikesaintsg** [reply]
What about a `--starter` flag? Might get more use out of a flag like that on your end than us in userland. In case there's any other small differences when it's a starter kit

**Comment by Julien-R44** [comment]
The inertia starter kit will now add the route on `/` rather than `/inertia`, starting from version `1.0.0-30`. See https://github.com/adonisjs/inertia/commit/71b8e5e999108fda7a4449e5fb1e432345aafc6b
